### ISSUE TITLE
nsp output is garbled

### ISSUE THREAD

**Issue by doug-wade** [nodesecurity/nsp#100]
I'm getting invalid output from nsp on [node v6](https://nodejs.org/en/blog/release/v6.0.0/).  Note that there were significant changes to the Buffer API.  From the blog post:

> Buffer
> New Buffer constructors have been added #4682 and #5833.
> Existing Buffer() and SlowBuffer() constructors have been deprecated in docs #4682 and #5833.
> Previously deprecated Buffer APIs are removed #5048, #4594.
> Improved error handling #4514.
> The Buffer.prototype.lastIndexOf() method has been added #4846.

Example output
```
dougwade npm-experiments/nsp-cli » npm init --yes
Wrote to /Users/doug.wade/npm-experiments/nsp-cli/package.json:

{
  "name": "nsp-cli",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}
dougwade npm-experiments/nsp-cli » npm i -S sinopa -q
dougwade npm-experiments/nsp-cli » nsp check
(+) 3 vulnerabilities found
┌───────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Quoteless Attributes in Templates can lead to Content Injection                                                                                         │
├───────────────┼──────────────────────────────────────────────────────────�%                                                                                                     dougwade npm-experiments/nsp-cli » node -v                                                                                                                                    1 ↵
v6.0.0
```

I think this is the root cause in a failure in gulp-nsp that looks like this

```
[14:15:40] 'nsp' errored after 4.53 s
[14:15:40] Error in plugin 'gulp-nsp'
Message:
    (+) Debug output: {"type":"Buffer","data":[60,104,116,109,108,62,60,98,111,100,121,62,60,104,49,62,53,48,51,32,83,101,114,118,105,99,101,32,85,110,97,118,97,105,108,97,98,108,101,60,47,104,49,62,10,78,111,32,115,101,114,118,101,114,32,105,115,32,97,118,97,105,108,97,98,108,101,32,116,111,32,104,97,110,100,108,101,32,116,104,105,115,32,114,101,113,117,101,115,116,46,10,60,47,98,111,100,121,62,60,47,104,116,109,108,62,10,10]}
(+) Error: Got an invalid response from Node Security, please email the above debug output to support@nodesecurity.io

npm ERR! Darwin 15.4.0
npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "run" "prepublish"
npm ERR! node v6.0.0
npm ERR! npm  v3.8.6
npm ERR! code ELIFECYCLE
npm ERR! @redfin/git-staged-util@0.0.0-7f2c42ef8edd80266bb9acc0daa2966ff67e2193 prepublish: `gulp prepublish`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @redfin/git-staged-util@0.0.0-7f2c42ef8edd80266bb9acc0daa2966ff67e2193 prepublish script 'gulp prepublish'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @redfin/git-staged-util package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp prepublish
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @redfin/git-staged-util
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @redfin/git-staged-util
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/doug.wade/code/main/redfin.npm/git-staged-util/npm-debug.log
[ERROR] Error: Command failed: npm run prepublish
    at checkExecSyncError (child_process.js:470:13)
    at execSync (child_process.js:510:13)
    at Object.run (/usr/local/lib/node_modules/rfnpm/lib/npm.js:114:5)
    at runScript (/usr/local/lib/node_modules/rfnpm/sub/run.js:31:7)
    at registry.walkGraph (/usr/local/lib/node_modules/rfnpm/sub/run.js:17:7)
    at walkGraph (/usr/local/lib/node_modules/rfnpm/lib/registry.js:169:3)
    at dependencies.forEach.depName (/usr/local/lib/node_modules/rfnpm/lib/registry.js:164:7)
    at Array.forEach (native)
    at walkGraph (/usr/local/lib/node_modules/rfnpm/lib/registry.js:161:16)
    at packages.forEach (/usr/local/lib/node_modules/rfnpm/lib/registry.js:48:7)
```

**Comment by evilpacket** [comment]
Thanks for the issue.  I was JUST looking into this and trying to figure out what the heck is going on. I was suspecting it was buffer related..... I'll post more when I know more.

**Comment by evilpacket** [comment]
Interesting to note that `nsp check -o json` also truncates output with node 6.0.0

**Comment by evanlucas** [comment]
https://github.com/nodesecurity/nsp/blob/master/lib/commands/check.js#L38 is the offender. stdout/stderr is not flushed when process.exit() is called. It would be better to set `process.exitCode = code` and then just return to make sure they are properly flushed.

**Comment by evilpacket** [comment]
@evanlucas thanks for the quick pr and sharp eye on that one.

**Comment by evanlucas** [comment]
happy to help!!

**Comment by evilpacket** [comment]
Just published `+ nsp@2.3.2`

**Comment by TheAlphaNerd** [comment]
@evilpacket I'm going to look into adding nsp to citgm so we can avoid surprises in the future
